src/node.cc

```diff
@@ -165,23 +168,21 @@ ArrayBufferAllocator ArrayBufferAllocator::the_singleton;
 
 
 void* ArrayBufferAllocator::Allocate(size_t length) {
-  if (length > kMaxLength)
+  void* data = malloc(length);
+  if (data == nullptr)
     return nullptr;
-  char* data = new char[length];
   memset(data, 0, length);
   return data;
 }
```

Comment:
I suggest replacing this with:

```c++
void* ArrayBufferAllocator::Allocate(size_t length) {
  return calloc(length, 1);
}
```

calloc() returns zeroed memory.  For larger allocations, libc can then just mmap the pages (newly mapped pages are all zero) without us zeroing it again.